### ISSUE TITLE
Error While Compiling: An Output Directory Must Be Specified

### ISSUE THREAD

**Issue by CalebGrove** [armin-pfaeffle/sass-autocompile#86]
When I compile my .scss file on save, it compiles properly, but throws this error:

> Command failed: node-sass --output-style compressed sass/main.scss main.css
> An output directory must be specified when compiling a directory

What's odd is that I have no `sass/main.scss` file in my project. This doesn't happen when I use the key combo (⌃⇧C), only when I save (⌘S).

**Comment by armin-pfaeffle** [comment]
What is the filename of your file?
Do you have a first-line-comment?

**Comment by CalebGrove** [comment]
No, I do not have a first-line-comment. Is that required? I didn't see anything about it being required in the docs. The filename is `grove-design.scss`. It's just a simple, single, SCSS file with no dependancies.

**Comment by armin-pfaeffle** [comment]
No, you don't need a first-line-comment. It's for a custom configuration, more or less. If you don't have one, your `grove-design.scss` should compile without an errors, in case the option `Compile on Save` is true.

I wonder why there is such an error message... Can you send me your scss file? Perhaps there is a problem in it?!

**Comment by CalebGrove** [comment]
I have uninstalled and reinstalled sass-autocompile, and that appears to have fixed the problem! Thanks for the awesome package, I'll report back if the problem reappears.

**Comment by armin-pfaeffle** [comment]
Great that it works now!

**Comment by scootman** [comment]
I am very new to sass (and atom) - and I'm getting the same error on save (relating to main.scss). Unistalled and re-installed package but still get error.

**Comment by armin-pfaeffle** [comment]
Hey! Sorry for my later answer. Had lot of important things to do. Is the problem still relevant?

Well, I don't know why that error occurs. Do you use a Mac, too? Can you try to put this in your first line:
```
// compileCompressed: ./main.css
```
Or can you try to set an absolute path like `/home/Users/username/.../ main.css`?

**Comment by amacfie** [comment]
I had this issue and it turned out that the script within which I ran sass didn't change to the proper directory first.

**Comment by armin-pfaeffle** [comment]
What exactly do you mean with that? Can you send me your files?

**Comment by amacfie** [comment]
I mean that the problem turned out to be the fault of neither `node-sass` nor `sass-autocompile` but of a custom script. Confusion may arise because the error message "An output directory must be specified when compiling a directory" is given by `node-sass` if its input file doesn't exist (even though it is not being called on a directory).

**Comment by mintysam** [comment]
I had the same issue and changing the command line to the following fixed the issue. (My files reside inside the src folder.)

"command": "node-sass ./src/style.scss ./src/style.css",

**Comment by zppro** [comment]
I'm sure it's the node-sass cli param [input file] is not set corretly

**Comment by yann-yinn** [comment]
> the error message "An output directory must be specified when compiling a directory" is given by node-sass if its input file doesn't exist (even though it is not being called on a directory).

thanks so much @amacfie, that was my error. node-sass error seems very confusing in this situation.

**Comment by blachawk** [comment]
I solved my issue.  The following ended up working for me and helps me understad that the "output directory" relates to the path in which the styles.scss is referred to.  So the following helped me.  Plus I was one directory deep...


```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "tasks": [
        {
            "label": "Sass Compile",
            "type": "shell",
            "command": "node-sass", 
            "args" : ["./project01/styles.scss", "./project01/styles.css"],
            "group":"build"
        }
    ]
}
```

**Comment by okyr-ux** [comment]
@mintysam, could you please explain your solution in more details? Or anyone?

I've tried to reinstall the package, to compile with the comment and add the path to the package settings (Include paths field). What is really strange — yesterday everything was working like a charm!

**UPD:** okay, in my case there was a '$' symbol in the project path. Now it works.

**Comment by armin-pfaeffle** [comment]
@blachawk: You run node-sass via terminal? And what is this for a configuration file? For a build system?

If this is the case, it has nothing to do with the mentioned problem.

**Comment by blachawk** [comment]
@armin-pfaeffle I removed my comment so that clarity can be focused on the given issue.

**Comment by anthonyhubs** [comment]
Hello;
I'm hoping that someone can offer advice to me.  I have the same problem as above.
Computer OS is linux-light with newest version of sass, ruby, nodejs, atom and newly re-installed sass-autocompile 0.13.5
In editing a .sass file, the changes started applying to the min.css and now they don't.  With every save, the 'output directory must be specified when compiling a directory' error still appears.
I saw mintysam's suggestion, but don't know which command line should be changed. Is that a change to one of the autocompile source files?
Thank you in advance for any suggestions

**Comment by armin-pfaeffle** [comment]
Can you try to compile a minimal file?
```SCSS
// compileCompressed: ./$1.min.css

$color: red;
body {
    background-color: $color;
}
```
Save that file via Atom. What happens?

Else: try do compile the file on your own by opening the terminal, switching to the corresponding path and entierung this command:
```
node-sass --output-style compressed test.scss > test.min.css
```
Adjust the filenames!

What happens here?

**Comment by anthonyhubs** [comment]
Thank you Armin.
When I did compile a minimal file, the error showed up - Command failed: node-sass --output-style compressed undefined/./css/sass//main.scss undefined/./css//main.css
An output directory must be specified when compiling a directory (as with the others here, main.css is not the filename)
However the .css did in fact update (added the change to my .css and also created a min.css.  I deleted the text within the .css, wrote a couple more css commands and saved again to confirm.  So rather then being non-functional, it's just a mild annoyance.
(can't confirm if the terminal command did the change because Atom did it)

**Comment by armin-pfaeffle** [comment]
Well... I really don't know where the `main.scss` comes from. I think there must be a bug in the path-building-process of my package but I cannot reproduce this problem.

@Supercalafragalisticexpialadoshus: Can you help me hunting down this bug? I added some logging command to the main script file of my package. Can you replace the current version and send me the output?

Here the instructions:
1. Close Atom.
2. Rename `/home/<user>/.atom/packages/sass-autocompile/lib/compiler.coffee` to e.g. `/home/<user>/.atom/packages/sass-autocompile/lib/compiler.coffee.backup`
3. Save attached `compiler.coffee` to `/home/<user>/.atom/packages/sass-autocompile/lib/` ([compiler.coffee.zip](https://github.com/armin-pfaeffle/sass-autocompile/files/2115586/compiler.coffee.zip)).
4. Open Atom.
5. Enable **Developer Tools** by selecting the following items in main menu: `View` > `Developer` > `Toggle Developer Tools`
6. Open **Console** tab in Developer Tools.
7. Open a SASS/SCSS file which includes a first-line-comment, so compilation is executed on save and save this file.
8. Open each Object output in the console tab by clicking on the arrow infront of `Object`, so details are visible.
9. Copy the complete output of the console tab und post it here. If you think that there are privacy problems you can send me an email: mail@armin-pfaeffle.de
10. Close Atom and restore the original `compiler.coffee`, see 11. and 12..
11. Delete `/home/<user>/.atom/packages/sass-autocompile/lib/compiler.coffee`.
12. Rename `/home/<user>/.atom/packages/sass-autocompile/lib/compiler.coffee.backup` to `/home/<user>/.atom/packages/sass-autocompile/lib/compiler.coffee`

**// Edit:** Sorry, I've forgotten the attachment...

**Comment by armin-pfaeffle** [comment]
The problem described by @Supercalafragalisticexpialadoshus was another package that displayed that error message. So, the problem has been fixed.

**Comment by jeturcotte** [comment]
I'll add on that I got the above error when the SOURCE file was misnamed in the command; e.g., ./sass/mystyles.scss  when in fact I'd renamed it and hadn't updated my copy and paste from the boilerplate instructions (via bulma's how-to.)  The error message just wasn't specific enough to point me in the correct direction at first.

**Comment by yonatanLehman** [comment]
In my case, the error was because the filename has uppercase letters (in windows 10). Changing the filename to all lowercase solved the problem.  it seems that node-sass does not accept file names with uppercase in them ?

**Comment by nagasaijithin** [comment]
> I'm sure it's the node-sass cli param [input file] is not set corretly

how to solve it

**Comment by megalon** [comment]
I am new to SASS and I was able to fix this issue when I noticed that SASS files use the extension `.scss` not `.sass` like I thought.
That typo is very easy to miss in a command, and it will cause node-sass to throw this error.

**Comment by nonacosa** [comment]
@megalon  omg, thanks 😆

**Comment by Evavic44** [comment]
Had this error just recently. The problem as earlier stated was a path linking issue. Had to specify the right folder where my `SASS` and `CSS` files live. 

```js
"sass": "node-sass -w src/scss/ -o src/dist/css/ --recursive"
```

Where `src` is the folder where your `SASS` and `CSS` files are.
